### ISSUE TITLE
MD053 False Positive when between two dollar currencies.

### ISSUE THREAD

**Issue by MartyGentillon** [DavidAnson/markdownlint#1734]
- What did you do?

I ran markdownlint (in visual studio code) against the folowing:

```
$n million [^12]. $m million.

[^12]: foobar
```
- What did you expect to happen?

The code to validate.

- What actually happened?

MD053 triggered.

- What messages or errors were there?

```
MD053/link-image-reference-definitions: Link and image reference definitions should be needed [Unused link or image reference definition: "^12"]markdownlint[MD053](https://github.com/DavidAnson/markdownlint/blob/v0.38.0/doc/md053.md)
```

- How can the issue be reproduced?
By attempting to validate the above.

- What version were you using?

The VSCode extension 
markdownlint v0.60.0

- What operating system were you using?
Windows

**Comment by DavidAnson** [comment]
The two inline dollar signs create a math block which you can see in the rendered output with the demo app. Escaping the first dollar sign will avoid the misrecognition and resolve your issue.

https://dlaa.me/markdownlint/#%25m%23%20Issue%201734%0A%0A%24n%20million%20%5B%5E12%5D.%20%24m%20million.%0A%0A%5B%5E12%5D%3A%20foobar%0A

**Comment by MartyGentillon** [comment]
Except that most implementations of inline math blocks that I am aware of forbid spaces after the initial $ or before the final one.  Likely for reasons related to this very issue.  Perhaps there needs to be configuration for that?

**Comment by MartyGentillon** [comment]
Obsidian, at least, does not interpret said text as being a math block.

**Comment by MartyGentillon** [comment]
See also: https://docs.zettlr.com/en/core/math/

**Comment by DavidAnson** [comment]
Here is a link to the parser plugin being used: https://github.com/micromark/micromark-extension-math

Per the "Syntax" section there, it does not appear to forbid leading/trailing spaces in the math span. If you disagree with that, please open an issue in their repository. 

However, based on scanning their closed issues, I think they will reject that suggestion and claim that matching the existing backtick behavior for code spans offers better consistency.

Consider:
```
text ` code ` text
```

**Comment by MartyGentillon** [comment]
Thanks
